vendor/github.com/jacobsa/gcloud/gcs/bucket.go

```diff
@@ -357,11 +359,40 @@ func newBucket(
 	url *url.URL,
 	userAgent string,
 	name string,
-	billingProject string) (b Bucket, err error) {
+	billingProject string,
+	tokenSrc oauth2.TokenSource,
+	goClientConfig *GoClientConfig) (b Bucket, err error) {
 
 	// Creating client through Go Storage Client Library for the storageClient parameter of bucket.
+	var tr *http.Transport = nil
+
+	// Choosing between HTTP1 and HTTP2.
+	if goClientConfig.EnableHTTP1 {
+
+		// Disables HTTP 2.0.
+		tr = &http.Transport{
+			MaxConnsPerHost:     goClientConfig.MaxConnsPerHost,
+			MaxIdleConnsPerHost: goClientConfig.MaxIdleConnsPerHost,
+			TLSNextProto: make(
+				map[string]func(string, *tls.Conn) http.RoundTripper,
+			),
+		}
+	} else {
+		tr = &http.Transport{
+			DisableKeepAlives: goClientConfig.DisableKeepAlives,
+			MaxConnsPerHost:   goClientConfig.MaxConnsPerHost,
+			ForceAttemptHTTP2: goClientConfig.ForceAttemptHTTP2,
```

Comment:
forceAttemptHttp2 and disableKeepAlives i dont see a need for them to be defined as flags. We are sure http2 performance improved only by setting these flags. So when enableHttp1 is false, we should set these flags to true by default